kintoclient/__init__.py

```diff
@@ -0,0 +1,254 @@
+        resp.raise_for_status()
+        return resp.json(), resp.headers
+
+
+class Permissions(object):
+    """Handles the permissions as sets"""
+    def __init__(self, container, permissions=None):
+        containers = CONTAINER_PERMISSIONS.keys()
+        if container not in containers:
+            msg = 'container should be one of %s' % ','.join(containers)
+            raise AttributeError(msg)
+
+        if permissions is None:
+            permissions = {}
+
+        self.container = container
+        self.permissions = permissions
+
+        for permission_type in CONTAINER_PERMISSIONS[container]:
+            attr = permission_type.replace(':', '_')
+            setattr(self, attr, set(permissions.get(permission_type, set())))
+
+    def serialize(self):
+        """Serialize the permissions to be sent to the server"""
+        to_save = {}
+        for permission_type in CONTAINER_PERMISSIONS[self.container]:
+            attr = permission_type.replace(':', '_')
+            to_save[permission_type] = getattr(self, attr)
+        return to_save
+
+
+class Bucket(object):
+    """
+    All operations are rooted in a bucket. It makes little sense for
+    one application to handle multiple buckets at once.
+    """
+
+    def __init__(self, name, server_url=None, auth=None, session=None,
+                 create=False):
+        """
+        :param name:
+            The name of the bucket to retrieve.
+        :param server_url:
+            The URL of the server to use.
+        :param auth:
+            A requests authentication policy object.
+        :param session:
+            An optional session object to use, rather than creating a new one.
+        :param create:
+            Defines if the bucket should be created. (default to False)
+        """
+        self.session = create_session(server_url, auth, session)
+        self.name = name
+
+        method = 'put' if create and name != 'default' else 'get'
+        self.uri = '/buckets/%s' % self.name
+        info, _ = self.session.request(method, self.uri)
+
+        self.data = info['data']
+        self.permissions = Permissions(
+            container='bucket',
+            permissions=info.get('permissions'))
+
+    def _get_collection_uri(self, collection_id):
+        return '%s/collections/%s' % (self.uri, collection_id)
+
+    def get_collection(self, name):
+        return Collection(name, bucket=self, session=self.session)
+
+    def list_collections(self):
+        uri = "%s/%s" % (self.uri, 'collections')
+        resp, _ = self.session.request('get', uri)
+
+        return [collection['id'] for collection in resp['data']]
+
+    def create_collection(self, name, permissions=None):
+        return Collection(name, bucket=self, session=self.session,
+                          create=True, permissions=permissions)
+
+    def delete_collection(self, name):
+        uri = self._get_collection_uri(name)
+        self.session.request('delete', uri)
```

Comment:
Since the server returns content on delete, I believe the client could also return it